lib/LedgerSMB/File.pm

```diff
@@ -306,7 +307,7 @@ sub DEMOLISH {
        unlink $self->{file_path} . '/' . $file;
    }
    closedir (TMP);
-   rmdir $self->{file_path};
+   return rmdir $self->{file_path};
```

Comment:
In the face of 25% test coverage, usually no doc of the return values, and no response on lsmb.dev, I mostly preserved the actual logic of the code, rather than go with my guess of the author's intention.  20% bad guesses * 20% bad luck on those * 25% covered leaves a couple of unnoticed bugs alive in code.  

I varied from conservative to very conservative on risking logic changes.    I did change loop-condition-returns and where an ending conditional gave very disparate final values.

So I agree with all your 'probably meant "return;"', but used a more conservative standard.
I have no problem with redoing any or all of these on your say so.